\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={PICO analysis},
            pdfauthor={Kim Dill-McFarland},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{PICO analysis}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Kim Dill-McFarland}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{version August 14, 2018}


\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\section{Setup}\label{setup}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{Sys.info}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                                                                                           sysname 
##                                                                                          "Darwin" 
##                                                                                           release 
##                                                                                          "17.7.0" 
##                                                                                           version 
## "Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64" 
##                                                                                          nodename 
##                                                                    "ubcmicros-MBP.hitronhub.home" 
##                                                                                           machine 
##                                                                                          "x86_64" 
##                                                                                             login 
##                                                                                             "kim" 
##                                                                                              user 
##                                                                                             "kim" 
##                                                                                    effective_user 
##                                                                                             "kim"
\end{verbatim}

Set seed for reproducibility

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{set.seed}\NormalTok{(}\DecValTok{2486}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Packages}\label{packages}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Community Ecology Package: Ordination, Diversity and Dissimilarities}
\KeywordTok{library}\NormalTok{(vegan)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: permute
\end{verbatim}

\begin{verbatim}
## Loading required package: lattice
\end{verbatim}

\begin{verbatim}
## This is vegan 2.5-2
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Handling and analysis of high-throughput phylogenetic sequence data.}
\KeywordTok{library}\NormalTok{(phyloseq)}
\CommentTok{#Data manipulation and plotting}
\KeywordTok{library}\NormalTok{(tidyverse)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## -- Attaching packages ------------------------------------------ tidyverse 1.2.1 --
\end{verbatim}

\begin{verbatim}
## √ ggplot2 2.2.1     √ purrr   0.2.5
## √ tibble  1.4.2     √ dplyr   0.7.5
## √ tidyr   0.8.1     √ stringr 1.3.1
## √ readr   1.1.1     √ forcats 0.3.0
\end{verbatim}

\begin{verbatim}
## -- Conflicts --------------------------------------------- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Nonparametric Tests for Repeated Measures Data in Factorial Designs}
\KeywordTok{library}\NormalTok{(nparLD)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: MASS
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'MASS'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:dplyr':
## 
##     select
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Fisheries stock assessment methods and data}
\CommentTok{#Dunn's Kruskal-Wallis Multiple Comparisons}
\KeywordTok{library}\NormalTok{(FSA)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## ## FSA v0.8.20. See citation('FSA') if used in publication.
## ## Run fishR() for related website and fishR('IFAR') for related book.
\end{verbatim}

\subsection{Data}\label{data}

\subsubsection{Import data}\label{import-data}

Import microbiome data at OTU, genus, and family levels

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU =}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/PICO.OTU.txt"}\NormalTok{, }\DataTypeTok{header=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{row.names=}\DecValTok{1}\NormalTok{, }\DataTypeTok{sep=}\StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\NormalTok{genus =}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/Genus.txt"}\NormalTok{, }\DataTypeTok{header=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{row.names=}\DecValTok{1}\NormalTok{, }\DataTypeTok{sep=}\StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\NormalTok{family =}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/Family.txt"}\NormalTok{, }\DataTypeTok{header=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{row.names=}\DecValTok{1}\NormalTok{, }\DataTypeTok{sep=}\StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Import full taxonomies of OTUs

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tax =}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/PICO.tax.taxonomy"}\NormalTok{, }\DataTypeTok{header=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{sep=}\StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\DataTypeTok{row.names=}\DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Import metadata of samples including alpha-diversity measures

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{meta =}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/PICO.meta.txt"}\NormalTok{, }\DataTypeTok{header=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{row.names=}\DecValTok{1}\NormalTok{, }\DataTypeTok{sep=}\StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Sort data}\label{sort-data}

Sort data by sample names.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU =}\StringTok{ }\NormalTok{OTU[}\KeywordTok{order}\NormalTok{(}\KeywordTok{row.names}\NormalTok{(OTU)),]}
\NormalTok{genus =}\StringTok{ }\NormalTok{genus[(}\KeywordTok{row.names}\NormalTok{(genus)),]}
\NormalTok{family =}\StringTok{ }\NormalTok{family[(}\KeywordTok{row.names}\NormalTok{(family)),]}
\NormalTok{meta =}\StringTok{ }\NormalTok{meta[}\KeywordTok{order}\NormalTok{(}\KeywordTok{row.names}\NormalTok{(meta)),]}
\end{Highlighting}
\end{Shaded}

\subsubsection{Create variables}\label{create-variables}

Create a pairwise variable for Treatment:Time\_point

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time =}\StringTok{ }\KeywordTok{paste}\NormalTok{(meta}\OperatorTok{$}\NormalTok{Treatment, meta}\OperatorTok{$}\NormalTok{Time_Point, }\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Functions}\label{functions}

Load necessary custom functions

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Function to run pairwise PERMANOVAs of all levels of a variable of interest across beta-diversity measures}
\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/pairwise_adonis_all.R"}\NormalTok{)}

\CommentTok{#Similarity percentages (SIMPER) analysis of multiple variables}
\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/simper.R"}\NormalTok{)}

\CommentTok{#Nonparametric Tests for Repeated Measures Data in Factorial Designs across multiple taxa}
\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/nparLD.R"}\NormalTok{)}

\CommentTok{#Dunn's Kruskal-Wallis Multiple Comparisons across multiple taxa}
\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/dunn.R"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\section{Beta-diversity}\label{beta-diversity}

\subsection{Calculate}\label{calculate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{BC.dist=}\KeywordTok{as.matrix}\NormalTok{(}\KeywordTok{vegdist}\NormalTok{(OTU, }\DataTypeTok{distance=}\StringTok{"bray"}\NormalTok{))}
\NormalTok{J.dist=}\KeywordTok{as.matrix}\NormalTok{(}\KeywordTok{vegdist}\NormalTok{(OTU, }\DataTypeTok{distance=}\StringTok{"jaccard"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\subsection{PERMANOVA}\label{permanova}

Model: Treatment*Time\_Point with Treatment:Time\_Point removed if not
significant

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{adonis}\NormalTok{(BC.dist }\OperatorTok{~}\StringTok{ }\NormalTok{Treatment}\OperatorTok{+}\KeywordTok{as.factor}\NormalTok{(Time_Point), }\DataTypeTok{strata=}\KeywordTok{as.factor}\NormalTok{(meta}\OperatorTok{$}\NormalTok{Subject_ID), }\DataTypeTok{data =}\NormalTok{ meta, }\DataTypeTok{permutations =} \DecValTok{1000}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## adonis(formula = BC.dist ~ Treatment + as.factor(Time_Point),      data = meta, permutations = 1000, strata = as.factor(meta$Subject_ID)) 
## 
## Blocks:  strata 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##                       Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)    
## Treatment              1    0.6961 0.69611  1.8343 0.02218 0.000999 ***
## as.factor(Time_Point)  2    1.0819 0.54093  1.4254 0.03448 0.000999 ***
## Residuals             78   29.6006 0.37949         0.94334             
## Total                 81   31.3785                 1.00000             
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{adonis}\NormalTok{(J.dist }\OperatorTok{~}\StringTok{ }\NormalTok{Treatment}\OperatorTok{+}\KeywordTok{as.factor}\NormalTok{(Time_Point), }\DataTypeTok{strata=}\KeywordTok{as.factor}\NormalTok{(meta}\OperatorTok{$}\NormalTok{Subject_ID), }\DataTypeTok{data =}\NormalTok{ meta, }\DataTypeTok{permutations =} \DecValTok{1000}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## adonis(formula = J.dist ~ Treatment + as.factor(Time_Point),      data = meta, permutations = 1000, strata = as.factor(meta$Subject_ID)) 
## 
## Blocks:  strata 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##                       Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)    
## Treatment              1    0.6961 0.69611  1.8343 0.02218 0.000999 ***
## as.factor(Time_Point)  2    1.0819 0.54093  1.4254 0.03448 0.000999 ***
## Residuals             78   29.6006 0.37949         0.94334             
## Total                 81   31.3785                 1.00000             
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
\end{verbatim}

\subsection{Pairwise PERMANOVA}\label{pairwise-permanova}

Model: Pairwise within Treatment:Time\_Point groups

Within treatment, across time, strata by subject

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{pairwise.adonis.all}\NormalTok{(OTU, meta, }\DataTypeTok{factors=}\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time, }\DataTypeTok{sim.method=}\StringTok{"bray"}\NormalTok{, }\DataTypeTok{stratify=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{strata.factor=}\NormalTok{meta}\OperatorTok{$}\NormalTok{Subject_ID, }\DataTypeTok{p.adjust.m=}\StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
##                     pairs   F.Model         R2 p.value p.adjusted
## 1  Placebo_0 vs Placebo_4 0.8960113 0.03599015   0.255     0.3060
## 2  Placebo_0 vs Placebo_8 1.1009292 0.04567995   0.028     0.0840
## 3     Placebo_0 vs Drug_0 0.9849814 0.03790580      NA         NA
## 4     Placebo_0 vs Drug_4 1.1548310 0.04415364      NA         NA
## 5     Placebo_0 vs Drug_8 1.9307970 0.07744626      NA         NA
## 6  Placebo_4 vs Placebo_8 0.7266190 0.02824386   0.077     0.1365
## 7     Placebo_4 vs Drug_0 1.5851333 0.05545307      NA         NA
## 8     Placebo_4 vs Drug_4 1.0585043 0.03772490      NA         NA
## 9     Placebo_4 vs Drug_8 1.8686655 0.06954813      NA         NA
## 10    Placebo_8 vs Drug_0 1.2400912 0.04552449      NA         NA
## 11    Placebo_8 vs Drug_4 0.6025125 0.02264871      NA         NA
## 12    Placebo_8 vs Drug_8 0.7658045 0.03092185      NA         NA
## 13       Drug_0 vs Drug_4 1.1429399 0.03921841   0.091     0.1365
## 14       Drug_0 vs Drug_8 1.2684822 0.04651826   0.027     0.0840
## 15       Drug_4 vs Drug_8 0.5401157 0.02035092   0.583     0.5830
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{pairwise.adonis.all}\NormalTok{(OTU, meta, }\DataTypeTok{factors=}\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time, }\DataTypeTok{sim.method=}\StringTok{"jaccard"}\NormalTok{, }\DataTypeTok{stratify=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{strata.factor=}\NormalTok{meta}\OperatorTok{$}\NormalTok{Subject_ID, }\DataTypeTok{p.adjust.m=}\StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
## 'nperm' >= set of all permutations: complete enumeration.
\end{verbatim}

\begin{verbatim}
## Set of permutations < 'minperm'. Generating entire set.
## Set of permutations < 'minperm'. Generating entire set.
\end{verbatim}

\begin{verbatim}
##                     pairs   F.Model         R2 p.value p.adjusted
## 1  Placebo_0 vs Placebo_4 0.8960113 0.03599015   0.278     0.3336
## 2  Placebo_0 vs Placebo_8 1.1009292 0.04567995   0.025     0.1050
## 3     Placebo_0 vs Drug_0 0.9849814 0.03790580      NA         NA
## 4     Placebo_0 vs Drug_4 1.1548310 0.04415364      NA         NA
## 5     Placebo_0 vs Drug_8 1.9307970 0.07744626      NA         NA
## 6  Placebo_4 vs Placebo_8 0.7266190 0.02824386   0.071     0.1420
## 7     Placebo_4 vs Drug_0 1.5851333 0.05545307      NA         NA
## 8     Placebo_4 vs Drug_4 1.0585043 0.03772490      NA         NA
## 9     Placebo_4 vs Drug_8 1.8686655 0.06954813      NA         NA
## 10    Placebo_8 vs Drug_0 1.2400912 0.04552449      NA         NA
## 11    Placebo_8 vs Drug_4 0.6025125 0.02264871      NA         NA
## 12    Placebo_8 vs Drug_8 0.7658045 0.03092185      NA         NA
## 13       Drug_0 vs Drug_4 1.1429399 0.03921841   0.116     0.1740
## 14       Drug_0 vs Drug_8 1.2684822 0.04651826   0.035     0.1050
## 15       Drug_4 vs Drug_8 0.5401157 0.02035092   0.549     0.5490
\end{verbatim}

Within time, between treatments

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{pairwise.adonis.all}\NormalTok{(OTU, meta, }\DataTypeTok{factors=}\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time, }\DataTypeTok{sim.method=}\StringTok{"bray"}\NormalTok{, }\DataTypeTok{stratify=}\OtherTok{FALSE}\NormalTok{, }\DataTypeTok{p.adjust.m=}\StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                     pairs   F.Model         R2 p.value p.adjusted
## 1  Placebo_0 vs Placebo_4 0.8960113 0.03599015   0.583  0.7950000
## 2  Placebo_0 vs Placebo_8 1.1009292 0.04567995   0.320  0.6000000
## 3     Placebo_0 vs Drug_0 0.9849814 0.03790580   0.471  0.7065000
## 4     Placebo_0 vs Drug_4 1.1548310 0.04415364   0.224  0.5442857
## 5     Placebo_0 vs Drug_8 1.9307970 0.07744626   0.009  0.1350000
## 6  Placebo_4 vs Placebo_8 0.7266190 0.02824386   0.770  0.9426923
## 7     Placebo_4 vs Drug_0 1.5851333 0.05545307   0.044  0.2200000
## 8     Placebo_4 vs Drug_4 1.0585043 0.03772490   0.383  0.6383333
## 9     Placebo_4 vs Drug_8 1.8686655 0.06954813   0.022  0.1650000
## 10    Placebo_8 vs Drug_0 1.2400912 0.04552449   0.125  0.4687500
## 11    Placebo_8 vs Drug_4 0.6025125 0.02264871   0.981  0.9810000
## 12    Placebo_8 vs Drug_8 0.7658045 0.03092185   0.817  0.9426923
## 13       Drug_0 vs Drug_4 1.1429399 0.03921841   0.254  0.5442857
## 14       Drug_0 vs Drug_8 1.2684822 0.04651826   0.160  0.4800000
## 15       Drug_4 vs Drug_8 0.5401157 0.02035092   0.969  0.9810000
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{pairwise.adonis.all}\NormalTok{(OTU, meta, }\DataTypeTok{factors=}\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time, }\DataTypeTok{sim.method=}\StringTok{"jaccard"}\NormalTok{, }\DataTypeTok{stratify=}\OtherTok{FALSE}\NormalTok{, }\DataTypeTok{p.adjust.m=}\StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                     pairs   F.Model         R2 p.value p.adjusted
## 1  Placebo_0 vs Placebo_4 0.8960113 0.03599015   0.563  0.7677273
## 2  Placebo_0 vs Placebo_8 1.1009292 0.04567995   0.300  0.5625000
## 3     Placebo_0 vs Drug_0 0.9849814 0.03790580   0.478  0.7170000
## 4     Placebo_0 vs Drug_4 1.1548310 0.04415364   0.207  0.5175000
## 5     Placebo_0 vs Drug_8 1.9307970 0.07744626   0.007  0.1050000
## 6  Placebo_4 vs Placebo_8 0.7266190 0.02824386   0.785  0.9461538
## 7     Placebo_4 vs Drug_0 1.5851333 0.05545307   0.024  0.1700000
## 8     Placebo_4 vs Drug_4 1.0585043 0.03772490   0.373  0.6216667
## 9     Placebo_4 vs Drug_8 1.8686655 0.06954813   0.034  0.1700000
## 10    Placebo_8 vs Drug_0 1.2400912 0.04552449   0.134  0.4500000
## 11    Placebo_8 vs Drug_4 0.6025125 0.02264871   0.979  0.9790000
## 12    Placebo_8 vs Drug_8 0.7658045 0.03092185   0.820  0.9461538
## 13       Drug_0 vs Drug_4 1.1429399 0.03921841   0.258  0.5528571
## 14       Drug_0 vs Drug_8 1.2684822 0.04651826   0.150  0.4500000
## 15       Drug_4 vs Drug_8 0.5401157 0.02035092   0.960  0.9790000
\end{verbatim}

\subsection{SIMPER}\label{simper}

Similarity percentages of Treatment:Time\_Point at OTU, genus, and
family levels

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{simper.pretty}\NormalTok{(}\DataTypeTok{x=}\NormalTok{OTU, }\DataTypeTok{metrics=}\NormalTok{meta, }\DataTypeTok{interesting=}\StringTok{"treat_time"}\NormalTok{, }\DataTypeTok{perc_cutoff=}\DecValTok{1}\NormalTok{, }\DataTypeTok{low_cutoff=}\StringTok{'n'}\NormalTok{, }\DataTypeTok{output_name=}\StringTok{"simper/OTU"}\NormalTok{)}

\KeywordTok{simper.pretty}\NormalTok{(}\DataTypeTok{x=}\NormalTok{genus, }\DataTypeTok{metrics=}\NormalTok{meta, }\DataTypeTok{interesting=}\StringTok{"treat_time"}\NormalTok{, }\DataTypeTok{perc_cutoff=}\DecValTok{1}\NormalTok{, }\DataTypeTok{low_cutoff=}\StringTok{'n'}\NormalTok{, }\DataTypeTok{low_val=}\FloatTok{0.001}\NormalTok{, }\DataTypeTok{output_name=}\StringTok{"simper/Genus"}\NormalTok{)}

\KeywordTok{simper.pretty}\NormalTok{(}\DataTypeTok{x=}\NormalTok{family, }\DataTypeTok{metrics=}\NormalTok{meta, }\DataTypeTok{interesting=}\StringTok{"treat_time"}\NormalTok{, }\DataTypeTok{perc_cutoff=}\DecValTok{1}\NormalTok{, }\DataTypeTok{low_cutoff=}\StringTok{'n'}\NormalTok{, }\DataTypeTok{output_name=}\StringTok{"simper/Family"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\section{Taxa of interest}\label{taxa-of-interest}

\subsection{ANOVA-like nparLD}\label{anova-like-nparld}

Nonparametric Tests for the F1-LD-F1 Design for taxa selected in SIMPER

For OTUs

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU.list =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"Otu00001"}\NormalTok{,}\StringTok{"Otu00002"}\NormalTok{,}\StringTok{"Otu00003"}\NormalTok{,}\StringTok{"Otu00004"}\NormalTok{,}\StringTok{"Otu00005"}\NormalTok{,}\StringTok{"Otu00006"}\NormalTok{,}\StringTok{"Otu00007"}\NormalTok{,}\StringTok{"Otu00008"}\NormalTok{,}\StringTok{"Otu00009"}\NormalTok{,}\StringTok{"Otu00010"}\NormalTok{,}\StringTok{"Otu00011"}\NormalTok{,}\StringTok{"Otu00012"}\NormalTok{,}\StringTok{"Otu00013"}\NormalTok{,}\StringTok{"Otu00014"}\NormalTok{,}\StringTok{"Otu00015"}\NormalTok{,}\StringTok{"Otu00016"}\NormalTok{,}\StringTok{"Otu00017"}\NormalTok{,}\StringTok{"Otu00019"}\NormalTok{,}\StringTok{"Otu00023"}\NormalTok{,}\StringTok{"Otu00024"}\NormalTok{,}\StringTok{"Otu00025"}\NormalTok{,}\StringTok{"Otu00027"}\NormalTok{,}\StringTok{"Otu00032"}\NormalTok{)}

\KeywordTok{nparLD.pval.fxn}\NormalTok{(OTU, OTU.list, }\StringTok{"nparLD/OTU"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For genera

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genus.list =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"g__Akkermansia"}\NormalTok{,}\StringTok{"g__Bacteroides"}\NormalTok{,}\StringTok{"g__Bifidobacterium"}\NormalTok{,}\StringTok{"g__Blautia"}\NormalTok{,}\StringTok{"g__Clostridium"}\NormalTok{,}\StringTok{"g__Collinsella"}\NormalTok{,}\StringTok{"g__Coprococcus"}\NormalTok{,}\StringTok{"g__Dorea"}\NormalTok{,}\StringTok{"g__Enterococcus"}\NormalTok{,}\StringTok{"g__Escherichia"}\NormalTok{,}\StringTok{"g__Faecalibacterium"}\NormalTok{,}\StringTok{"g__Lactobacillus"}\NormalTok{,}\StringTok{"g__Prevotella"}\NormalTok{,}\StringTok{"g__Roseburia"}\NormalTok{,}\StringTok{"g__Ruminococcus_L"}\NormalTok{,}\StringTok{"g__Ruminococcus_R"}\NormalTok{,}\StringTok{"g__Streptococcus"}\NormalTok{,}\StringTok{"g__Veillonella"}\NormalTok{)}

\KeywordTok{nparLD.pval.fxn}\NormalTok{(genus, genus.list, }\StringTok{"nparLD/genus"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For families

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fam.list =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"f__Bifidobacteriaceae"}\NormalTok{,}\StringTok{"f__Clostridiaceae"}\NormalTok{,}\StringTok{"f__Enterobacteriaceae"}\NormalTok{,}\StringTok{"f__Enterococcaceae"}\NormalTok{,}\StringTok{"f__Lachnospiraceae"}\NormalTok{,}\StringTok{"f__Lactobacillaceae"}\NormalTok{,}\StringTok{"f__Ruminococcaceae"}\NormalTok{,}\StringTok{"f__Streptococcaceae"}\NormalTok{,}\StringTok{"f__Veillonellaceae"}\NormalTok{,}\StringTok{"f__Verrucomicrobiaceae"}\NormalTok{)}

\KeywordTok{nparLD.pval.fxn}\NormalTok{(family, fam.list, }\StringTok{"nparLD/family"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Merge results}\label{merge-results}

Read in results

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU.nparLD.pval =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"nparLD/OTU.nparLD.csv"}\NormalTok{)}
\NormalTok{genus.nparLD.pval =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"nparLD/genus.nparLD.csv"}\NormalTok{)}
\NormalTok{fam.nparLD.pval =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"nparLD/family.nparLD.csv"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Merge, reformat, and save

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{nparLD.pval =}\StringTok{ }\NormalTok{OTU.nparLD.pval }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(genus.nparLD.pval) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(fam.nparLD.pval) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{filter}\NormalTok{(test}\OperatorTok{==}\StringTok{"aov"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(}\OperatorTok{-}\NormalTok{test) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{group_by}\NormalTok{(taxon) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{spread}\NormalTok{(}\DataTypeTok{key=}\NormalTok{variable, }\DataTypeTok{value=}\NormalTok{pval)}

\KeywordTok{write.csv}\NormalTok{(nparLD.pval, }\StringTok{"nparLD/all.nparLD.pval.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{FDR correction}\label{fdr-correction}

FDR correct nparLD p-values

For OTUs

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU.nparLD.fdr =}\StringTok{ }\NormalTok{OTU.nparLD.pval }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Combing statistical test and variable into 1 term and remove originals}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{group=}\KeywordTok{paste}\NormalTok{(test, variable, }\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(}\OperatorTok{-}\NormalTok{test, }\OperatorTok{-}\NormalTok{variable) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Reformat to wide format}
\StringTok{  }\KeywordTok{spread}\NormalTok{(}\DataTypeTok{key=}\NormalTok{group, }\DataTypeTok{value=}\NormalTok{pval) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# FDR correct pvals for each test_variable}
\StringTok{  }\KeywordTok{mutate_if}\NormalTok{(is.numeric, }\KeywordTok{funs}\NormalTok{(}\DataTypeTok{fdr =} \KeywordTok{p.adjust}\NormalTok{(., }\DataTypeTok{method=}\StringTok{'fdr'}\NormalTok{))) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Remove original pvals}
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(taxon, }\KeywordTok{contains}\NormalTok{(}\StringTok{"fdr"}\NormalTok{))}

\KeywordTok{write.csv}\NormalTok{(OTU.nparLD.fdr, }\StringTok{"nparLD/OTU.nparLD.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For genera

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genus.nparLD.fdr =}\StringTok{ }\NormalTok{genus.nparLD.pval }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Combing statistical test and variable into 1 term and remove originals}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{group=}\KeywordTok{paste}\NormalTok{(test, variable, }\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(}\OperatorTok{-}\NormalTok{test, }\OperatorTok{-}\NormalTok{variable) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Reformat to wide format}
\StringTok{  }\KeywordTok{spread}\NormalTok{(}\DataTypeTok{key=}\NormalTok{group, }\DataTypeTok{value=}\NormalTok{pval) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# FDR correct pvals for each test_variable}
\StringTok{  }\KeywordTok{mutate_if}\NormalTok{(is.numeric, }\KeywordTok{funs}\NormalTok{(}\DataTypeTok{fdr =} \KeywordTok{p.adjust}\NormalTok{(., }\DataTypeTok{method=}\StringTok{'fdr'}\NormalTok{))) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Remove original pvals}
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(taxon, }\KeywordTok{contains}\NormalTok{(}\StringTok{"fdr"}\NormalTok{))}

\KeywordTok{write.csv}\NormalTok{(genus.nparLD.fdr, }\StringTok{"nparLD/genus.nparLD.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For families

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fam.nparLD.fdr =}\StringTok{ }\NormalTok{fam.nparLD.pval }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Combing statistical test and variable into 1 term and remove originals}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{group=}\KeywordTok{paste}\NormalTok{(test, variable, }\DataTypeTok{sep=}\StringTok{"_"}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(}\OperatorTok{-}\NormalTok{test, }\OperatorTok{-}\NormalTok{variable) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Reformat to wide format}
\StringTok{  }\KeywordTok{spread}\NormalTok{(}\DataTypeTok{key=}\NormalTok{group, }\DataTypeTok{value=}\NormalTok{pval) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# FDR correct pvals for each test_variable}
\StringTok{  }\KeywordTok{mutate_if}\NormalTok{(is.numeric, }\KeywordTok{funs}\NormalTok{(}\DataTypeTok{fdr =} \KeywordTok{p.adjust}\NormalTok{(., }\DataTypeTok{method=}\StringTok{'fdr'}\NormalTok{))) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{# Remove original pvals}
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(taxon, }\KeywordTok{contains}\NormalTok{(}\StringTok{"fdr"}\NormalTok{))}

\KeywordTok{write.csv}\NormalTok{(fam.nparLD.fdr, }\StringTok{"nparLD/family.nparLD.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Merge results}\label{merge-results-1}

Read in and merge FDR corrected nparLD p-values

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{nparLD.fdr =}\StringTok{ }\NormalTok{OTU.nparLD.fdr }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(genus.nparLD.fdr) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(fam.nparLD.fdr) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(taxon, }\KeywordTok{starts_with}\NormalTok{(}\StringTok{"aov_"}\NormalTok{)) }

\KeywordTok{write.csv}\NormalTok{(nparLD.fdr, }\StringTok{"nparLD/all.nparLD.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Pairwise Dunn}\label{pairwise-dunn}

Dunn's Kruskal-Wallis Multiple Comparisons for taxa selected in SIMPER

For OTUs

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU.list2=}\KeywordTok{c}\NormalTok{(}\StringTok{"Otu00002"}\NormalTok{, }\StringTok{"Otu00003"}\NormalTok{, }\StringTok{"Otu00004"}\NormalTok{, }\StringTok{"Otu00005"}\NormalTok{, }\StringTok{"Otu00006"}\NormalTok{, }\StringTok{"Otu00008"}\NormalTok{, }\StringTok{"Otu00014"}\NormalTok{, }\StringTok{"Otu00015"}\NormalTok{, }\StringTok{"Otu00017"}\NormalTok{, }\StringTok{"Otu00024"}\NormalTok{, }\StringTok{"Otu00025"}\NormalTok{, }\StringTok{"Otu00027"}\NormalTok{)}

\KeywordTok{dunn.pval.fxn}\NormalTok{(OTU, OTU.list2, }\StringTok{"dunn/OTU"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For genera

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genus.list2 =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"g__Bacteroides"}\NormalTok{, }\StringTok{"g__Collinsella"}\NormalTok{, }\StringTok{"g__Enterococcus"}\NormalTok{, }\StringTok{"g__Escherichia"}\NormalTok{, }\StringTok{"g__Roseburia"}\NormalTok{, }\StringTok{"g__Ruminococcus_L"}\NormalTok{, }\StringTok{"g__Akkermansia"}\NormalTok{, }\StringTok{"g__Clostridium"}\NormalTok{, }\StringTok{"g__Lactobacillus"}\NormalTok{)}

\KeywordTok{dunn.pval.fxn}\NormalTok{(genus, genus.list2, }\StringTok{"dunn/genus"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For families

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{family.list2=}\KeywordTok{c}\NormalTok{(}\StringTok{"f__Enterobacteriaceae"}\NormalTok{, }\StringTok{"f__Enterococcaceae"}\NormalTok{, }\StringTok{"f__Lachnospiraceae"}\NormalTok{, }\StringTok{"f__Verrucomicrobiaceae"}\NormalTok{)}

\KeywordTok{dunn.pval.fxn}\NormalTok{(family, family.list2, }\StringTok{"dunn/family"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Merge results}\label{merge-results-2}

Read in Dunn results

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{OTU.dunn.fdr =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"dunn/OTU.dunn.fdr.csv"}\NormalTok{)}
\NormalTok{genus.dunn.fdr =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"dunn/genus.dunn.fdr.csv"}\NormalTok{)}
\NormalTok{fam.dunn.fdr =}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"dunn/family.dunn.fdr.csv"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Merge, select comparisons of interest, and save

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dunn.fdr =}\StringTok{ }\NormalTok{OTU.dunn.fdr }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(genus.dunn.fdr) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rbind}\NormalTok{(fam.dunn.fdr) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{filter}\NormalTok{(}\OperatorTok{!}\NormalTok{(label }\OperatorTok{%in%}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"0_Drug - 4_Placebo"}\NormalTok{, }
                        \StringTok{"0_Drug - 8_Placebo"}\NormalTok{, }
                        \StringTok{"4_Placebo - 8_Drug"}\NormalTok{, }
                        \StringTok{"0_Placebo - 4_Drug"}\NormalTok{,}
                        \StringTok{"0_Placebo - 8_Drug"}\NormalTok{,}
                        \StringTok{"4_Drug - 8_Placebo"}\NormalTok{))) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{group_by}\NormalTok{(taxon) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{spread}\NormalTok{(}\DataTypeTok{key=}\NormalTok{label, }\DataTypeTok{value=}\NormalTok{fdr_pval)}

\KeywordTok{write.csv}\NormalTok{(dunn.fdr, }\StringTok{"dunn/all.dunn.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Summarise results}\label{summarise-results}

Merge all nparLD and Dunn p-values.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tax.temp =}\StringTok{ }\NormalTok{tax }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{rownames_to_column}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rename}\NormalTok{(}\DataTypeTok{taxon=}\NormalTok{rowname) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(taxon, Family, Genus)}

\NormalTok{all.fdr =}\StringTok{ }\NormalTok{nparLD.fdr }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{right_join}\NormalTok{(dunn.fdr, }\DataTypeTok{by=}\StringTok{"taxon"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{left_join}\NormalTok{(tax.temp, }\DataTypeTok{by=}\StringTok{"taxon"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Column `taxon` joining factors with different levels, coercing to
## character vector
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{write.csv}\NormalTok{(all.fdr, }\StringTok{"all.npar.dunn.fdr.csv"}\NormalTok{, }\DataTypeTok{row.names=}\OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Mean +/- s.d.}\label{mean---s.d.}

Determine mean and standard deviation of abundances of taxa of interest
within Treatment:Time\_Point groups

Calculate abundances

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gen.abund =}\StringTok{ }\NormalTok{genus}\OperatorTok{/}\KeywordTok{rowSums}\NormalTok{(genus)}
\NormalTok{fam.abund =}\StringTok{ }\NormalTok{family}\OperatorTok{/}\KeywordTok{rowSums}\NormalTok{(family)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Bacteroides}\label{bacteroides}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.935748
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.653574
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.515562
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.900627
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.625566
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3.581518
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{###}

\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.241711
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 6.891377
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.08733937
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.1880266
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.995586
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Bacteroides}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.444368
\end{verbatim}

\subsubsection{Ruminococcus\_L}\label{ruminococcus_l}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 5.473395
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.33377
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.037039
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3.364184
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 11.3261
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 14.18201
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{###}

\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.472326
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 8.437927
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.541525
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 9.787987
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 13.78263
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(gen.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{g__Ruminococcus_L}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 19.00898
\end{verbatim}

\subsubsection{Verrucomicrobiaceae}\label{verrucomicrobiaceae}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.532012
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 14.38547
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.19313
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 10.16118
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3.75033
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Placebo"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 10.01492
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{###}

\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.8472772
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{0} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.636724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.1234638
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{4} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.3220946
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.474631
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(fam.abund[meta}\OperatorTok{$}\NormalTok{Time_Point }\OperatorTok{==}\StringTok{ }\DecValTok{8} \OperatorTok{&}\StringTok{ }\NormalTok{meta}\OperatorTok{$}\NormalTok{Treatment }\OperatorTok{==}\StringTok{ "Drug"}\NormalTok{ ,]}\OperatorTok{$}\NormalTok{f__Verrucomicrobiaceae}\OperatorTok{*}\DecValTok{100}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.90139
\end{verbatim}

\section{Alpha-diversity}\label{alpha-diversity}

\subsection{Normality}\label{normality}

Check normalilty of Shannon's diveristy metric

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plot}\NormalTok{(}\KeywordTok{density}\NormalTok{(meta}\OperatorTok{$}\NormalTok{shannon))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PICO_analysis_files/figure-latex/normal-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Plot using a qqplot}
\KeywordTok{qqnorm}\NormalTok{(meta}\OperatorTok{$}\NormalTok{shannon)}
\KeywordTok{qqline}\NormalTok{(meta}\OperatorTok{$}\NormalTok{shannon, }\DataTypeTok{col =} \DecValTok{2}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PICO_analysis_files/figure-latex/normal-2.pdf}

\subsection{ANOVA}\label{anova}

Model: Treatment*Time\_Point with Treatment:Time\_Point removed if not
significant

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{summary}\NormalTok{(}\KeywordTok{aov}\NormalTok{(shannon }\OperatorTok{~}\StringTok{ }\NormalTok{Treatment }\OperatorTok{+}\StringTok{ }\KeywordTok{as.factor}\NormalTok{(Time_Point) }\OperatorTok{+}\StringTok{ }\KeywordTok{Error}\NormalTok{(Subject_ID), }\DataTypeTok{data=}\NormalTok{meta))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Error: Subject_ID
##           Df    Sum Sq   Mean Sq
## Treatment  1 8.727e-05 8.727e-05
## 
## Error: Within
##                       Df Sum Sq Mean Sq F value Pr(>F)  
## Treatment              1   3.20   3.195   4.685 0.0335 *
## as.factor(Time_Point)  2   1.33   0.667   0.978 0.3807  
## Residuals             77  52.51   0.682                 
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
\end{verbatim}

\subsection{Pairwise t-test}\label{pairwise-t-test}

Model: Pairwise within Treatment:Time\_Point groups

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{pairwise.t.test}\NormalTok{(meta}\OperatorTok{$}\NormalTok{shannon, meta}\OperatorTok{$}\NormalTok{treat_time, }\DataTypeTok{p.adjust.method =}\StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Pairwise comparisons using t tests with pooled SD 
## 
## data:  meta$shannon and meta$treat_time 
## 
##           Drug_0 Drug_4 Drug_8 Placebo_0 Placebo_4
## Drug_4    0.660  -      -      -         -        
## Drug_8    0.886  0.779  -      -         -        
## Placebo_0 0.064  0.146  0.096  -         -        
## Placebo_4 0.039  0.100  0.063  0.897     -        
## Placebo_8 0.720  0.948  0.836  0.143     0.098    
## 
## P value adjustment method: none
\end{verbatim}

\subsection{FDR correction}\label{fdr-correction-1}

FDR correct alpha-diversity p-values

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{p.adjust}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\FloatTok{0.660}\NormalTok{,}\FloatTok{0.886}\NormalTok{,}\FloatTok{0.779}\NormalTok{,}\FloatTok{0.897}\NormalTok{,}\FloatTok{0.143}\NormalTok{,}\FloatTok{0.098}\NormalTok{,}\FloatTok{0.064}\NormalTok{,}\FloatTok{0.100}\NormalTok{,}\FloatTok{0.836}\NormalTok{), }\DataTypeTok{method=}\StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.89700 0.89700 0.89700 0.89700 0.32175 0.30000 0.30000 0.30000 0.89700
\end{verbatim}

\section{Figures}\label{figures}

Order Treatment:Time\_Point groups

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{meta}\OperatorTok{$}\NormalTok{treat_time_ord =}\StringTok{ }\KeywordTok{factor}\NormalTok{(meta}\OperatorTok{$}\NormalTok{treat_time,}\KeywordTok{c}\NormalTok{(}\StringTok{"Placebo_0"}\NormalTok{,}\StringTok{"Drug_0"}\NormalTok{,}\StringTok{"Placebo_4"}\NormalTok{,}\StringTok{"Drug_4"}\NormalTok{,}\StringTok{"Placebo_8"}\NormalTok{,}\StringTok{"Drug_8"}\NormalTok{))  }
\end{Highlighting}
\end{Shaded}

\subsection{Alpha boxplot}\label{alpha-boxplot}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{ggplot}\NormalTok{(meta, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x=}\NormalTok{treat_time_ord, }\DataTypeTok{y=}\NormalTok{shannon)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_boxplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_bw}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{labs}\NormalTok{(}\DataTypeTok{y =} \StringTok{"Shannon's diveristy index"}\NormalTok{, }\DataTypeTok{x =} \StringTok{""}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PICO_analysis_files/figure-latex/alpha.plot-1.pdf}

\subsection{Taxa boxplots}\label{taxa-boxplots}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Select family of interest}
\NormalTok{fam.temp =}\StringTok{ }\NormalTok{fam.abund }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rownames_to_column}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(rowname, f__Verrucomicrobiaceae)}

\CommentTok{#Select genera of interest}
\NormalTok{gen.temp =}\StringTok{ }\NormalTok{gen.abund }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rownames_to_column}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(rowname, g__Bacteroides, g__Ruminococcus_L) }

\CommentTok{#merge with metadata}
\NormalTok{meta }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rownames_to_column}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(rowname, treat_time, treat_time_ord) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{full_join}\NormalTok{(fam.temp, }\DataTypeTok{by=}\StringTok{"rowname"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{full_join}\NormalTok{(gen.temp, }\DataTypeTok{by=}\StringTok{"rowname"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{rename}\NormalTok{(}\DataTypeTok{Sample=}\NormalTok{rowname) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\CommentTok{#reformat to long}
\StringTok{  }\KeywordTok{gather}\NormalTok{(}\DataTypeTok{key=}\StringTok{"taxon"}\NormalTok{, }\DataTypeTok{value=}\StringTok{"abund"}\NormalTok{, }\OperatorTok{-}\NormalTok{Sample, }\OperatorTok{-}\NormalTok{treat_time, }\OperatorTok{-}\NormalTok{treat_time_ord) }\OperatorTok{%>%}\StringTok{ }

\StringTok{  }\CommentTok{#plot}
\KeywordTok{ggplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_boxplot}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x=}\NormalTok{treat_time_ord, }\DataTypeTok{y=}\NormalTok{abund)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_bw}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{labs}\NormalTok{(}\DataTypeTok{y =} \StringTok{"Relative abundance (%)"}\NormalTok{, }\DataTypeTok{x =} \StringTok{""}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_grid}\NormalTok{(taxon}\OperatorTok{~}\NormalTok{.)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PICO_analysis_files/figure-latex/taxa.plot-1.pdf}

\section{FIN}\label{fin}


\end{document}
